### PR TITLE
[build][pytorch][graviton]graviton pytorch image build

### PR DESCRIPTION
**Pull request by Qingzi-Lan**
### Description
* add graviton pytorch docker file
    * currently using place holder pytorch binary
* Ensure that MME is possible with Graviton endpoints on SageMaker when the feature is ready. If MME is not available for Graviton endpoints, remove the capabilities.multi-models LABEL from the dockerfile.
### Tests run
* pulled image and ran inference test

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Qingzi-Lan** [comment]
sanity and eks test fails due to image and test instance platform incompatible.

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton [hunk 1]
```diff
@@ -1,4 +1,139 @@
...
 
-# Placeholder Dockerfile for Graviton
-RUN apt-get update
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
+
+ARG PYTHON=python3
+ARG PYTHON_VERSION=3.8.10
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/Dockerfile.graviton:9]: Can we use 3.8.12?

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton [hunk 2]
```diff
@@ -1,4 +1,139 @@
...
+    requests \
+ && conda clean -ya \
+ && pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3 \
+ && pip install packaging==20.4 \
+    enum-compat==0.0.3 \
+    "cryptography>=3.3.2"
+
+# install torch and torchvision from the PyTorch website
+RUN pip install torch torchvision
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/Dockerfile.graviton:99]: Will these be replaced later with AWS forks of the torch and torchvision packages?

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton [hunk 3]
```diff
@@ -1,4 +1,139 @@
...
+
+RUN conda install -y -c conda-forge "pyyaml>=5.4,<5.5"
+RUN pip install pillow==8.2.0 "awscli<2"
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/Dockerfile.graviton:102]: Can we use Pillow 8.3.2 here?

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton [hunk 4]
```diff
@@ -1,4 +1,139 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.8/license.txt -o /license.txt
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/Dockerfile.graviton:135]: Please change this to https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.9/license.txt

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton [hunk 5]
```diff
@@ -1,4 +1,139 @@
...
+
+EXPOSE 8080 8081
+ENTRYPOINT ["python", "/usr/local/bin/dockerd-entrypoint.py"]
+CMD ["torchserve", "--start", "--ts-config", "/home/model-server/config.properties", "--model-store", "/home/model-server/"]
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/Dockerfile.graviton:139]: nit: please add an empty line at the end of the file.

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton (hunk 2 above)
Comment (Qingzi-Lan) [reply pytorch/inference/docker/1.9/py3/Dockerfile.graviton:99]: yes, discussed this with Manu, will replace the wheel later.
current package from pip should be a good placeholder for all the tests

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton (hunk 3 above)
Comment (Qingzi-Lan) [reply pytorch/inference/docker/1.9/py3/Dockerfile.graviton:102]: will do

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton (hunk 5 above)
Comment (Qingzi-Lan) [reply pytorch/inference/docker/1.9/py3/Dockerfile.graviton:139]: will do.

File: pytorch/inference/docker/1.9/py3/Dockerfile.graviton [hunk 6]
```diff
@@ -1,4 +1,139 @@
-FROM ubuntu:18.04
+FROM arm64v8/ubuntu:20.04
 
-# Placeholder Dockerfile for Graviton
-RUN apt-get update
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/Dockerfile.graviton:6]: No changed needed here, but making a note on this PR for future reference:

Ensure that MME is possible with Graviton endpoints on SageMaker when the feature is ready. If MME is not available for Graviton endpoints, remove the `capabilities.multi-models` LABEL from the dockerfile.

**Review by saimidu — COMMENTED** [review]
Approved. Please revert the changes on the developer config toml file.

**Review by saimidu — APPROVED** [review]

**Pull request merged**